tests/compatibility-tests/src/main/resources/serial/serial.groovy

```diff
@@ -24,30 +24,46 @@ import org.apache.activemq.artemis.jms.client.*
 
 file = arg[0]
 method = arg[1]
-System.out.println("File::" + file);
+version = arg[2]
+System.out.println("File::" + file)
 
 
 if (method.equals("write")) {
     cf = new ActiveMQConnectionFactory("tcp://localhost:61616?confirmationWindowSize=1048576&blockOnDurableSend=false");
     queue = new ActiveMQQueue("queue");
     topic = new ActiveMQTopic("topic")
+    if (version.equals("ARTEMIS-SNAPSHOT")) {
+        destination = new ActiveMQDestination("address", "name", ActiveMQDestination.TYPE.DESTINATION, null)
```

Comment:
I told you to call another script as sometimes it fails the compilation. Perhaps this constructor won't be checked until you actually called it.


A lot of times I had to do this it involved classNames that didn't exist.. so I needed to a call an external script. 


I will wait the test running to be sure.